paasta_tools/marathon_tools.py

```diff
@@ -955,3 +956,18 @@ def get_healthcheck_for_instance(service, instance, service_manifest, random_por
         mode = None
         healthcheck_command = None
     return (mode, healthcheck_command)
+
+
+def kill_task(client, app_id, task_id, scale):
+    """ Wrapper to the official kill_task method that is idempotent """
+    try:
+        return client.kill_task(app_id=app_id, task_id=task_id, scale=True)
+    except MarathonHttpError as e:
+        # Marathon's interface is always async, so it is possible for you to see a task
+        # in the interface and kill it, yet by the time it tries to kill it, it is already
```

Comment:
I think the actual error is more along the lines of:

Marathon allows you to kill and scale in one action, but this is not idempotent. If you kill&scale the same task ID twice, the number of instances gets decremented twice. This can lead to a situation where kill&scaling the last task decrements the number of instances below zero, causing a "Bean is not valid" message.